web/assets/js/actions.ts

```diff
@@ -1,214 +1,391 @@
+                }
+            }, 1000);
+
+            setTimeout(() => {
+                // Clear the fetch interval
+                clearInterval(fetchInterval)
+
+                reject('wait timeout exceeded')
+            }, exitTimeout)
+        })
+    }
+
+    /**
+     * Fetches the active game state. Should be called
+     * within a loop and checked.
+     * @param gameID the ID of the running game
+     */
+    public async getGame(gameID: string): Promise<Game> {
+        const gameResponse: string = await this.provider?.evaluateExpression(
+            chessRealm,
+            `GetGame(${gameID})`
+        ) as string
+
+        // Parse the response
+        return JSON.parse(gameResponse)
+    }
+
+    /**
+     * Executes the move and returns the game state
+     * @param gameID the ID of the game
+     * @param from from position
+     * @param to  to position
+     * @param promotion promotion information
+     */
+    async makeMove(
+        gameID: string,
+        from: string,
+        to: string,
+        promotion: Promotion = Promotion.NO_PROMOTION,
+    ): Promise<Game> {
+        // Make the move
+        const moveResponse: BroadcastTxCommitResult = await this.wallet?.callMethod(
+            chessRealm,
+            "MakeMove",
+            [
+                gameID,
+                from,
+                to,
+                promotion.toString()
+            ],
+            TransactionEndpoint.BROADCAST_TX_COMMIT
+        ) as BroadcastTxCommitResult
+
+        // Parse the response from the node
+        const moveDataRaw: string | null = moveResponse.deliver_tx.ResponseBase.Data
+        if (!moveDataRaw) {
+            throw new Error("invalid move response")
+        }
+
+        // Magically parse the response
+        return JSON.parse(moveDataRaw)
+    }
+
+    /**
+     * Returns a flag indicating if the game with the specified ID is over
+     * @param gameID the ID of the running game
+     * @param type the game-over state types
+     */
+    async isGameOver(gameID: string, type: GameoverType): Promise<boolean> {
+        // Fetch the game state
+        const game: Game = await this.getGame(gameID)
+
+        return game.state === type
+    }
+
+    /**
+     * Triggers the game draw process
+     * @param gameID the ID of the running game
+     */
+    async requestDraw(gameID: string): Promise<Game> {
+        // Make the request
+        const drawResponse: BroadcastTxCommitResult = await this.wallet?.callMethod(
+            chessRealm,
+            "DrawOffer",
+            [
+                gameID,
+            ],
+            TransactionEndpoint.BROADCAST_TX_COMMIT
+        ) as BroadcastTxCommitResult
+
+        // Parse the response from the node
+        const drawResponseRaw: string | null = drawResponse.deliver_tx.ResponseBase.Data
+        if (!drawResponseRaw) {
+            throw new Error("invalid draw response")
+        }
+
```

Comment:
We need a timeout here (as before) to close the request when too long (9s before) by returning a `false`

Reply:
Added one:
cabbcc8d07159d472fc94d05db4f156059aac2cc

This will error out (effectively same result as false)